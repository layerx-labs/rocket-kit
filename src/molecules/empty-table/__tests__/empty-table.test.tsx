import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { EmptyTable } from '../../..';

describe('Empty Table', () => {
  it('renders', () => {
    const { asFragment } = render(
      <EmptyTable
        border
        tableHead={['Column 1', 'Column 2', 'Column 3', 'Column 4']}
        value="No data"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all columns', () => {
    render(
      <EmptyTable
        border
        tableHead={['Column 1', 'Column 2', 'Column 3', 'Column 4']}
        value="No data"
      />
    );
    for (let row = 1; row < 4; ++row) {
      screen.getByText(`Column ${row}`);
    }

    expect(screen.getByText(/no data/i)).toBeInTheDocument();
  });
});
