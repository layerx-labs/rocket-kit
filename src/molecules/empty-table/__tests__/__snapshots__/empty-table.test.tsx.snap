// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Table renders 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
}

.c2 {
  border-width: 0.0625rem;
  border-style: solid;
  border-color: #939393;
  border-radius: 0.375rem 0.375rem 0 0;
  height: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.85rem;
  font-weight: 700;
  color: #939393;
  text-transform: uppercase;
}

.c2 > div {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0 0.9375rem;
  font-size: 0.85rem;
  font-weight: 700;
  color: #939393;
  text-transform: uppercase;
}

.c2 > div:first-child,
.c2 > div:nth-child(2) {
  display: initial;
}

.c3 {
  border-width: 0 0.0625rem 0.0625rem 0.0625rem;
  border-style: solid;
  border-color: #939393;
  border-radius: 0 0 0.375rem 0.375rem;
}

.c4 {
  height: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4:not(:last-child) {
  border-bottom: 0.0625rem solid #939393;
}

.c4 > div {
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0 0.9375rem;
}

.c4 > div:first-child,
.c4 > div:nth-child(2) {
  display: inherit;
}

.c5 {
  background-color: #edeef1;
  width: 100%;
  height: 0.9375rem;
}

.c1 {
  position: absolute;
  bottom: 0;
  background-image: linear-gradient( to bottom, rgba(255,255,255,0), rgba(255,255,255,1) );
  width: 100%;
  height: calc(100% - 3.125rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding: 0 0.9375rem 0.9375rem 0.9375rem;
  text-align: center;
  color: #939393;
}

@media (min-width:700px) {
  .c2 > div:nth-child(3) {
    display: initial;
  }
}

@media (min-width:900px) {
  .c2 > div:nth-child(4) {
    display: initial;
  }
}

@media (min-width:1280px) {
  .c2 > div {
    display: initial;
  }
}

@media (min-width:700px) {
  .c4 > div:nth-child(3) {
    display: inherit;
  }
}

@media (min-width:900px) {
  .c4 > div:nth-child(4) {
    display: inherit;
  }
}

@media (min-width:1280px) {
  .c4 > div {
    display: inherit;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <span>
        No data
      </span>
    </div>
    <div
      class="c2"
    >
      <div>
        Column 1
      </div>
      <div>
        Column 2
      </div>
      <div>
        Column 3
      </div>
      <div>
        Column 4
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
      </div>
      <div
        class="c4"
      >
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
      </div>
      <div
        class="c4"
      >
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
        <div>
          <div
            class="c5"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
